### DISCUSSION TITLE
Pass ChannelReader from server to client via InvokeAsync fails with HubException: Client failed to parse Argument(s)

### DISCUSSION THREAD

**Discussion by bbartels** [dotnet/aspnetcore#58565] [Q&A]
Hi,
I am running into an issue when using `InvokeAsync` to stream data from a server to a client via a `ChannelReader`, where it is failing with `HubException: Client failed to parse Argument(s)`
I have a minimal repro here: https://github.com/bbartels/SignalRTest/tree/main

I couldn't find any docs specifically around my scenario. I have also previously tried to just use `SendDataAsync` with a `ChannelReader` more closely aligned with the scenarios in the docs: https://learn.microsoft.com/en-us/aspnet/core/signalr/streaming?view=aspnetcore-8.0, but didn't get anywhere either. While it didn't fail with an exception, it just did not invoke the handler.

In case it is relevant, what I want to achieve is a call to a single SignalR clients, stream some data to them, let them do some processing and stream some data back. `InvokeAsync` seems to suit itself well as it allows you to return a response. I am not sure if `ChannelReader` is allowed to be returned from `InvokeAsync`.

**Comment by BrennanConroy** [comment]
That is not supported. See https://github.com/dotnet/aspnetcore/issues/10151 and https://github.com/dotnet/aspnetcore/issues/46161

**Reply by bbartels** [reply]
Ah I see, that is unfortunate. Might be worth sticking something in the docs, I've been going mad haha
I appreciate the rapid response!
